File: go/000/TreeNode.go
package main

import "fmt"

type TreeNode struct {
    Val int
    Left *TreeNode
    Right *TreeNode
}

func getTreeNode(ints ...int) *TreeNode {
    tns := make([]*TreeNode, len(ints))
    for i, n := 0, len(ints); i < n; i++ {
        tns[i] = &TreeNode{ints[i], nil, nil}
    }
    for i, n := 0, len(ints); i < n; i++ {
        left, right := 2*i+1, 2*i+2
        if left >= len(tns) {break}
        tns[i].Left = tns[left]
        if right >= len(tns) {break}
        tns[i].Right = tns[right]
    }
    return tns[0]
}

func printTreeNode(nodes ...*TreeNode)  {
    for _, node := range nodes {
        p(node)
        fmt.Print("\n")
    }
}

func p(tn *TreeNode)  {
    if tn == nil {return}
    fmt.Print("[")
    p(tn.Left)
    fmt.Print(tn.Val)
    p(tn.Right)
    fmt.Print("]")
}
//go build TreeNode.go && TreeNode && go clean
func main()  {
    printTreeNode(getTreeNode(1, 5, 6, 2), getTreeNode(7, 8, 6, 2))
}

File: kt/000/TreeNode.kt

class TreeNode(var `val`: Int = 0) {
        var left: TreeNode? = null
        var right: TreeNode? = null

    companion object {

        fun getTreeNode(vararg `is`: Int): TreeNode {
            val tns = Array(`is`.size, {
                TreeNode(`is`[it])
            })
            for (i in tns.indices) {
                val left = 2 * i + 1
                val right = left + 1
                if (left >= tns.size) break
                tns[i].left = tns[left]
                if (right >= tns.size) break
                tns[i].right = tns[right]
            }
            return tns[0]
        }

        fun printTreeNode(vararg tns: TreeNode) {
            for (tn in tns) {
                print(tn)
                println()
            }
        }

        private fun print(tn: TreeNode?) {
            if (tn == null) return
            print('[')
            print(tn.left)
            print(tn.`val`)
            print(tn.right)
            print(']')
        }
    }
}
